util.go

```diff
@@ -0,0 +1,146 @@
+package sqlite
+
+import (
+	"context"
+	"database/sql"
+	"fmt"
+	"strings"
+)
+
+// DropAll deletes all the data from a database.
+func DropAll(ctx context.Context, conn *sql.Conn, schemaName string) (err error) {
+	defer func() {
+		if err != nil {
+			err = fmt.Errorf("sqlitedb.DropAll: %w", err)
+		}
+	}()
+
+	if schemaName == "" {
+		schemaName = "main"
+	}
+
+	var indexes, tables, triggers, views []string
+
+	rows, err := conn.QueryContext(ctx, fmt.Sprintf("SELECT name, type FROM %q.sqlite_schema", schemaName))
+	if err != nil {
+		return err
+	}
+	defer rows.Close()
+	for rows.Next() {
+		var name, sqlType string
+		if err := rows.Scan(&name, &sqlType); err != nil {
+			return err
+		}
+		switch sqlType {
+		case "index":
+			indexes = append(indexes, name)
+		case "table":
+			tables = append(tables, name)
+		case "trigger":
+			triggers = append(triggers, name)
+		case "view":
+			views = append(views, name)
+		default:
+			return fmt.Errorf("unknown sqlite schema type %q for %q", sqlType, name)
+		}
+	}
+	rows.Close()
+	if err := rows.Err(); err != nil {
+		return err
+	}
+
+	for _, name := range indexes {
+		if _, err := conn.ExecContext(ctx, fmt.Sprintf("DROP INDEX %q.%q", schemaName, name)); err != nil {
+			return err
+		}
+	}
+	for _, name := range triggers {
+		if _, err := conn.ExecContext(ctx, fmt.Sprintf("DROP TRIGGER %q.%q", schemaName, name)); err != nil {
+			return err
+		}
+	}
+	for _, name := range views {
+		if _, err := conn.ExecContext(ctx, fmt.Sprintf("DROP VIEW %q.%q", schemaName, name)); err != nil {
+			return err
+		}
+	}
+	for _, name := range tables {
+		if _, err := conn.ExecContext(ctx, fmt.Sprintf("DROP TABLE %q.%q", schemaName, name)); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+// CopyAll copies the contents of one database to another.
+//
+// Traditionally this is done in sqlite by closing the database and copying
+// the file. However it can be useful to do it online: a single exclusive
+// transaction can cross multiple databases, and if multiple processes are
+// using a file, this lets one replace the database without first
+// communicating with the other processes, asking them to close the DB first.
+func CopyAll(ctx context.Context, conn *sql.Conn, dstSchemaName, srcSchemaName string) (err error) {
```

Comment:
Document empty schema default

Reply:
documented that this matches sqlite's pragma schema-name